tests/benchmark/conftest.py

```diff
@@ -7,6 +7,32 @@
 
 def pytest_collection_modifyitems(config, items):
     """Add the `benchmark` marker to all tests under `./tests/benchmark`."""
-    for item in items:
-        if Path(__file__).parent in Path(item.fspath).parents:
-            item.add_marker(pytest.mark.benchmark)
+    marker_expr = config.getoption("-m", default="")
+    gas_benchmark_values = config.getoption("--gas-benchmark-values", default=None)
+    run_benchmarks = marker_expr and (
+        "benchmark" in marker_expr and "not benchmark" not in marker_expr
+    )
+
+    if gas_benchmark_values:
+        run_benchmarks = True
+
+    items_for_removal = []
+    for i, item in enumerate(items):
+        is_benchmark_test = Path(__file__).parent in Path(item.fspath).parents
```

Comment:
The test could also be a benchmark test if it already had the `pytest.mark.benchmark` (e.g. a benchmark-marked test outside of the `benchmark` folder).

This is not super well defined, but until we do I'd check for the marker too.